tests/src/rules/extensions.js

```diff
@@ -640,6 +640,24 @@ describe('TypeScript', () => {
             ],
             parser,
           }),
+          test({
+            code: 'import type T from "./typescript-declare";',
+            errors: ['Missing file extension for "./typescript-declare"'],
+            options: [
+              'always',
+              { ts: 'never', tsx: 'never', js: 'never', jsx: 'never', checkTypeImports: true },
+            ],
+            parser,
+          }),
+          test({
+            code: 'export type { MyType } from "./typescript-declare";',
+            errors: ['Missing file extension for "./typescript-declare"'],
+            options: [
+              'always',
+              { ts: 'never', tsx: 'never', js: 'never', jsx: 'never', checkTypeImports: true },
+            ],
+            parser,
+          }),
```

Comment:
Let's also add similar tests with it set to `always`

Reply:
See https://github.com/import-js/eslint-plugin-import/pull/2817/commits/6ac3d03217b80c068fc7846f77410ce5976f943f